src/main/java/org/prebid/server/bidder/globalsun/GlobalsunBidder.java

```diff
@@ -0,0 +1,141 @@
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.globalsun.ExtImpGlobalsun;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class GlobalsunBidder implements Bidder<BidRequest> {
+
+    private static final String TYPE_PUBLISHER = "publisher";
+
+    private static final TypeReference<ExtPrebid<?, ExtImpGlobalsun>> GLOBALSUN_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public GlobalsunBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public final Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<HttpRequest<BidRequest>> outgoingRequests = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            final ExtImpGlobalsun extImpGlobalsun;
+            try {
+                extImpGlobalsun = parseImpExt(imp);
+            } catch (PreBidException e) {
+                return Result.withError(BidderError.badInput(e.getMessage()));
+            }
+            outgoingRequests.add(createSingleRequest(modifyImp(imp, extImpGlobalsun), request));
+        }
+
+        return Result.withValues(outgoingRequests);
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.withValues(extractBids(httpCall.getRequest().getPayload(), bidResponse));
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private ExtImpGlobalsun parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), GLOBALSUN_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpGlobalsun extImpAndGlobalsun) {
```

Comment:
Pls put methods in call order
makeRequests
... makeRequest related
makeBids
...makeBids related